toolchain/check/handle_class.cpp

```diff
@@ -39,6 +60,25 @@ static auto BuildClassDecl(Context& context)
             context.insts().Get(existing_id).TryAs<SemIR::ClassDecl>()) {
       // This is a redeclaration of an existing class.
       class_decl.class_id = existing_class_decl->class_id;
+      auto& class_info = context.classes().Get(class_decl.class_id);
+
+      // The introducer kind must match the previous declaration.
+      // TODO: The rule here is not yet decided. See #3384.
+      if (class_info.inheritance_kind != inheritance_kind) {
+        CARBON_DIAGNOSTIC(
+            ClassRedeclarationDifferentIntroducer, Error,
+            "Class redeclared with a different inheritance kind.");
```

Comment:
```suggestion
            "Class redeclared with different inheritance kind.");
```

Should "a" be dropped here, for message terseness?